pkg/modifycontroller/controller_test.go

```diff
@@ -154,77 +110,152 @@ func TestModifyPVC(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			client := csi.NewMockClient("mock", true, true, true, true, true, false)
+			client := csi.NewMockClient(testDriverName, true, true, true, true, true, false)
 			if test.modifyFailure {
 				client.SetModifyFailed()
 			}
-			driverName, _ := client.GetDriverName(context.TODO())
 
-			initialObjects := []runtime.Object{}
-			if test.pvc != nil {
-				initialObjects = append(initialObjects, test.pvc)
+			initialObjects := []runtime.Object{test.pvc, test.pv, testVacObject, targetVacObject}
+			ctrlInstance, ctx := setupFakeK8sEnvironment(t, client, initialObjects)
+			defer ctx.Done()
+
+			_, _, err, _ := ctrlInstance.modify(test.pvc, test.pv)
+
+			if test.expectFailure && err == nil {
+				t.Errorf("for %s expected error got nothing", test.name)
 			}
-			if test.pv != nil {
-				test.pv.Spec.PersistentVolumeSource.CSI.Driver = driverName
-				initialObjects = append(initialObjects, test.pv)
+
+			if !test.expectFailure {
+				if err != nil {
+					t.Errorf("for %s, unexpected error: %v", test.name, err)
+				}
 			}
+		})
+	}
+}
 
-			// existing vac set in the pvc and pv
-			initialObjects = append(initialObjects, testVacObject)
-			// new vac used in modify volume
-			initialObjects = append(initialObjects, targetVacObject)
+func TestSyncPVC(t *testing.T) {
+	basePVC := createTestPVC(pvcName, targetVac /*vacName*/, testVac /*curVacName*/, testVac /*targetVacName*/)
+	basePV := createTestPV(1, pvcName, pvcNamespace, "foobaz" /*pvcUID*/, &fsVolumeMode, testVac)
 
-			kubeClient, informerFactory := fakeK8s(initialObjects)
-			pvInformer := informerFactory.Core().V1().PersistentVolumes()
-			pvcInformer := informerFactory.Core().V1().PersistentVolumeClaims()
-			vacInformer := informerFactory.Storage().V1beta1().VolumeAttributesClasses()
+	otherDriverPV := createTestPV(1, pvcName, pvcNamespace, "foobaz" /*pvcUID*/, &fsVolumeMode, testVac)
```

Comment:
This seems like a code smell. 

Happy to attempt to try to refactor resizer's testutils to implement Builder pattern to prevent by allowing developers to use `MakePVC().WithXXX().WithYYY()` style in the test-cases themselves, similar to what k/k has. 

This work was started in #402, but was pushed back to once all-in-one sidecar project is here.